airflow/ui/src/components/TriggerDag.tsx

```diff
@@ -0,0 +1,206 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  Box,
+  Button,
+  useDisclosure,
+  Text,
+  Accordion,
+  AccordionItem,
+  AccordionButton,
+  AccordionIcon,
+  AccordionPanel,
+  FormControl,
+  FormLabel,
+  Input,
+  FormErrorMessage,
+  Tr,
+  Td,
+  Table,
+} from "@chakra-ui/react";
+import { json, jsonParseLinter } from "@codemirror/lang-json";
+import { linter, lintGutter } from "@codemirror/lint";
+import { indentationMarkers } from "@replit/codemirror-indentation-markers";
+import CodeMirror, {
+  lineNumbers,
+  type ViewUpdate,
+} from "@uiw/react-codemirror";
+import { useForm, SubmitHandler, Controller } from "react-hook-form";
+import { FiPlay } from "react-icons/fi";
+
+import ActionModal from "./ActionModal";
+
+type Props = {
+  readonly dagDisplayName: string;
+  readonly dagId: string;
+};
+
+type TriggerDagModalProps = {
+  readonly isOpen: boolean;
+  readonly onClose: () => void;
+} & Props;
+
+const TriggerDagModal = ({
+  dagDisplayName,
+  dagId,
+  isOpen,
+  onClose,
+}: TriggerDagModalProps) => {
+  const onTrigger = (values: any) =>
+    new Promise<void>((resolve) => {
+      setTimeout(() => {
+        alert(JSON.stringify(values, null, 2));
+        resolve();
+        // Trigger here TODO
+        onClose();
+      }, 3000);
+    });
+
+  const {
+    control,
+    formState: { errors, isSubmitting },
+    handleSubmit,
+    register,
+  } = useForm();
+
+  return (
+    <ActionModal
+      header={`Trigger DAG ${dagDisplayName}`}
+      isOpen={isOpen}
+      onClose={onClose}
+      submitButton={
+        <Button isLoading={isSubmitting} onClick={handleSubmit(onTrigger)}>
```

Comment:
```suggestion
        <Button isLoading={isSubmitting} onClick={() => handleSubmit(onTrigger)}>
```

We need to pass the function not the return value of a called function.

Reply:
I tried this - also as eslint is cmplaining about this... but then it is not working...

Reply:
What error did you get? 

Sometimes I need to run `pnpm install` before linting.
In vscode, once in a while I also need to cmd+shift+P and restart the eslint server:
<img width="517" alt="Screenshot 2024-10-17 at 12 01 15 PM" src="https://github.com/user-attachments/assets/94696634-13d0-448d-b567-03cd038cebb8">